src/Filesystem.cc

```diff
@@ -36,466 +30,288 @@
-  removeFile(_newFilename, _warningOp);
+/////////////////////////////////////////////////
+std::string ignition::common::copyFromUnixPath(const std::string &_path)
+{
+  std::string copy = _path;
+  changeFromUnixPath(copy);
+  return copy;
+}
 
-  return false;
+/////////////////////////////////////////////////
+void ignition::common::changeToUnixPath(std::string &_path) {
+  std::replace(_path.begin(), _path.end(),
+      static_cast<char>(fs::path::preferred_separator), '/');
 }
 
 /////////////////////////////////////////////////
-std::string ignition::common::absPath(const std::string &_path)
+std::string ignition::common::copyToUnixPath(const std::string &_path)
 {
-  std::string result;
-
-  // cppcheck-suppress ConfigurationNotChecked
-  char path[IGN_PATH_MAX] = "";
-#ifdef _WIN32
-  if (GetFullPathName(_path.c_str(), IGN_PATH_MAX, &path[0], nullptr) != 0)
-#else
-  if (realpath(_path.c_str(), &path[0]) != nullptr)
-#endif
-    result = path;
-  else if (!_path.empty())
-  {
-    // If _path is an absolute path, then return _path.
-    // An absolute path on Windows is a character followed by a colon and a
-    // backslash.
-    if (_path.compare(0, 1, "/") == 0 || _path.compare(1, 3, ":\\") == 0)
-      result = _path;
-    // Otherwise return the current working directory with _path appended.
-    else
-      result = joinPaths(ignition::common::cwd(), _path);
-  }
-
-  ignition::common::replaceAll(result, result, "//", "/");
+  std::string copy = _path;
+  changeToUnixPath(copy);
+  return copy;;
+}
 
-  return result;
+/////////////////////////////////////////////////
+std::string ignition::common::absPath(const std::string &_path)
+{
+  return fs::absolute(_path).string();
 }
 
-// This is help function to handle windows paths, there are a mix between '/'
-// and backslashes.
-// joinPaths uses the system separator, in Windows this generate some issues
-// with URIs
-std::string checkWindowsPath(const std::string _path)
+/////////////////////////////////////////////////
+std::string ignition::common::joinPaths(
+    const std::string &_path1, const std::string &_path2)
 {
-  if (_path.empty())
-    return _path;
+  fs::path p1{_path1};
+  fs::path p2{_path2};
 
-  // Check if this is a http or https, if so change backslashes generated by
-  // jointPaths to '/'
-  if ((_path.size() > 7 && 0 == _path.compare(0, 7, "http://")) ||
-      (_path.size() > 8 && 0 == _path.compare(0, 8, "https://")))
-  {
-    return std::regex_replace(_path, std::regex(R"(\\)"), "/");
-  }
+  bool is_url = false;
+
+  if (_path1.find("://") == std::string::npos)
+    p1 = p1.lexically_normal();
+  else
+    is_url = true;
 
-  // This is a Windows path, convert all '/' into backslashes
-  std::string result = std::regex_replace(_path, std::regex(R"(/)"), "\\");
-  std::string drive_letters;
+  if (_path2.find("://") == std::string::npos)
+    p2 = p2.lexically_normal();
+  else
+    is_url = true;
```

Comment:
shouldn't it be an error if `_path2` is a URL? I guess unless `_path1` is empty. Maybe our existing tests don't address that, but it seems like a problem

Reply:
You are correct that it's strange.  It _likely_ should happen, but there is nothing stopping users from doing it.

I think I'd like to address this in a follow-up PR as it's out of the scope of our current test suite.

Reply:
can you add a todo comment before merging this?

Reply:
Done.